src/Piece.cpp

```diff
@@ -0,0 +1,81 @@
+#include "Piece.h"
+
+/**
+ * Sets the color of the piece.
+ * 
+ * @param color The color to set the piece to.
+ */
+void Piece::setIsWhite(bool color) { 
+	m_isWhite = color; 
+}
+
+/**
+ * @brief Gets the color of the piece.
+ *
+ * @return true if the piece is white, false otherwise.
+ */
+const bool Piece::getColor() const {
+    return m_isWhite;
+}
+
+void Piece::setPrevMove(const std::pair<std::pair<int, int>, std::pair<int, int>>& move) {
+    prevMove = move;
+}
+
+/**
+ * @brief Gets the previous move made by the piece.
+ *
+ * @return The previous move made by the piece.
+ */
+const std::pair<std::pair<int, int>, std::pair<int, int>>& Piece::getPrevMove() const {
+    return prevMove;
+}
+
+/**
+ * @brief Checks if the path of the move is clear of any pieces.
+ *
+ * @param moveCoords The start and end coordinates of the move.
+ * @param chessMatrix The current state of the chess board.
+ * @return true if the path is clear, false otherwise.
+ */
+const bool Piece::isPathClear(const std::pair<std::pair<int, int>, std::pair<int, int>>& moveCoords, const std::vector<std::vector<std::unique_ptr<Piece>>>& chessMatrix) const {
```

Comment:
can be simplified and made more readable by breaking it down into smaller functions with clear responsibilities. This can improve code maintainability and readability.

Reply:
Done.